security/cmd/node_agent/management/server.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/gogo/googleapis/google/rpc"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/security/cmd/node_agent/na"
+	"istio.io/istio/security/cmd/node_agent_k8s/workload/handler"
+	cagrpc "istio.io/istio/security/pkg/caclient/grpc"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/platform"
+	"istio.io/istio/security/pkg/workload"
+	pb "istio.io/istio/security/proto"
+)
+
+// Server specifies the node agent server.
+// TODO(incfly): adds sync.Mutex to protects the fields, such as `handlerMap`.
+type Server struct {
+	// map from uid to workload handler instance.
+	handlerMap map[string]handler.WorkloadHandler
+	// makes mgmt-api server to stop
+	done     chan bool
+	config   *na.Config
+	pc       platform.Client
+	cAClient cagrpc.CAGrpcClient
```

Comment:
Can we use CAClient instead of using pc+CAGrpcClient+createRequest()...?
https://github.com/istio/istio/blob/master/security/pkg/caclient/client.go
I guess that can save you some code :)

Reply:
Looked at the caclient code, it seems that it only supports single identity, obtaining identity in constructor and stores that as a private field. here we need support multi identity for diff. workloads. (I think it's mostly because the nodeagent by then only is scoped for vm mode?)

I added a TODO here and note in the #4190 to remove redundancy.